src/py/_pyodide/_core_docs.py

```diff
@@ -365,6 +368,72 @@ def to_js(
             js.Array.from -- convert to an array of entries
             js.Object.fromEntries -- convert to a JavaScript object
+    default_converter: Callable[[Any, Callable[[Any], JsProxy], Callable[[Any, JsProxy], None]], JsProxy], default=None
+        If present will be invoked whenever Pyodide does not have some built in
+        conversion for the object. If ``default_converter`` raises an error, the
+        error will be allowed to propagate. Otherwise, the object returned will
+        be used as the conversion. ``default_converter`` takes three arguments.
+        The first argument is the value to be converted.
+
+        Here are a couple examples of converter functions. In addition to the
+        normal conversions, convert ``Date`` to ``datetime``:
+
+        .. code-block:: python
+
+            from datetime import datetime
+            from js import Date
+            def default_converter(value, _ignored1, _ignored2):
+                if isinstance(value, datetime):
+                    return Date.new(value.timestamp() * 1000)
+                return value
+
+        Don't create any PyProxies, require a complete conversion or raise an error:
+
+        .. code-block:: python
+
+            def default_converter(_value, _ignored1, _ignored2):
+                raise Exception("Failed to completely convert object")
+
+        The second and third arguments are only needed for converting
+        containers. The second argument is a conversion function which is used
+        to convert the elements of the container with the same settings. The
+        third argument is a "cache" function which is needed to handle self
+        referential containers. Consider the following example. Suppose we have
+        a Python ``Pair`` class:
+
+        .. code-block:: python
+
+            class Pair:
+                def __init__(self, first, second):
+                    self.first = first
+                    self.second = second
+
+        We can use the following ``default_converter`` to convert ``Pair`` to ``Array``:
+
+        .. code-block:: python
+
+            from js import Array
+            def default_converter(value, convert, cache):
+                if not isinstance(value, Pair):
+                    return value
+                result = Array.new()
+                cache(value, result);
+                result.push(convert(value.first))
+                result.push(convert(value.second))
+                return result
+
+        Note that we have to cache the conversion of ``value`` before converting
+        ``value.first`` and ``value.second``. To see why, consider a self
+        referential pair:
+
+        .. code-block:: javascript
+
+            p = Pair(0, 0);
+            p.first = p;
+
+        Without ``cache(value, result);``, converting ``p`` would lead to an
+        infinite recurse. With it, we can successfully convert ``p`` to an Array
+        such that ``l[0] === l``.
```

Comment:
This might a bit long for a docstring. I wonder if we should move some of it to a user manual section in type translations about default conversions. Though it does seem very specific to this particular argument so not sure. Feel free not to do anything.